### PR TITLE
Replace MVTec with dummy dataset

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
# Notice

- Builds on top of https://github.com/openvinotoolkit/training_extensions/pull/664

# Changes

- Add dummy dataset

# Coverage

image

### CODE REVIEW

File: external/anomalib/tests/helpers/train.py [hunk 1]
```diff
@@ -58,8 +50,13 @@ class OTEAnomalyTrainer:
             Defaults to "./ote/configs/template.yaml".
     """
 
-    def __init__(self, model_template_path: str = "./ote/configs/template.yaml"):
-        dataset_path = os.path.join("./datasets/MVTec", "bottle")
+    def __init__(
+        self,
+        model_template_path: str = "./ote/configs/template.yaml",
```
Comment (samet-akcay) [inline external/anomalib/tests/helpers/train.py:55]: I know it doesn't change the overall behaviour of the test, but maybe we could change the default path here as well

File: external/anomalib/tests/helpers/dummy_dataset.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+                ) as dataset_path:
+                    kwds["category"] = "shapes"
+                    return func(*args, dataset_path=dataset_path, **kwds)
+
+        return inner
+
+
+class GeneratedDummyDataset(ContextDecorator):
+    """Context for generating dummy shapes dataset.
+    Example:
```
Comment (samet-akcay) [inline external/anomalib/tests/helpers/dummy_dataset.py:149]: there should be a newline before `Example`. I know we're not gonna run sphinx, but would be nice in terms of consistency.

Comment (samet-akcay) [reply external/anomalib/tests/helpers/dummy_dataset.py:149]: And same for the other stuff down below

File: external/anomalib/tests/helpers/shapes.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+from skimage.draw import polygon
+
+
+def random_square_patch(input_region: List[int], min_width: int = 10) -> List[int]:
+    """Gets a random patch in the input region
+
+    Args:
+        input_region (List[int]): Coordinates of the input region. [x1, y1, x2, y2]
+        min_width (int): Minimum width of the returned patch.
+    Example:
```
Comment (samet-akcay) [inline external/anomalib/tests/helpers/shapes.py:13]: Same

File: external/anomalib/tests/helpers/dummy_dataset.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+from .shapes import random_shapes
+
+
+def generate_random_anomaly_image(
+    image_width: int,
+    image_height: int,
+    shapes: List[str] = ["triangle", "rectangle"],
+    max_shapes: Optional[int] = 5,
+    generate_mask: Optional[bool] = False,
+) -> Dict:
```
Comment (samet-akcay) [inline external/anomalib/tests/helpers/dummy_dataset.py:39]: `Dict[str, np.ndarray]`?

File: external/anomalib/tests/helpers/dummy_dataset.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+        image_width (int): Width of the image
+        image_height (int): Height of the image
+        shapes (List[str]): List of shapes to draw in the image. Make sure these are different from `anomalous_shapes`
+        max_shapes (int): Maximum shapes of a kind in the image. Defaults to 5.
+        max_size (Optional[int], optional): Maximum size of the test shapes. Defaults to 10.
+        generate_mask (bool): Toggle between train/test split. Train images are restricted to first quadrant.
+                    Also generates the mask for the image. Defaults to False.
+
+    Returns:
+        Tuple: image if `train` is False. Else return image, mask
```
Comment (samet-akcay) [inline external/anomalib/tests/helpers/dummy_dataset.py:52]: Dict?

**Review by samet-akcay — COMMENTED** [review]
only a minor comment left for typing. Otherwise, it looks good

**Comment by ashwinvaidya17** [comment]
@samet-akcay @djdameln can we merge this? I have tested this against the CI http://ote.inn.intel.com:8080/job/hello_job/66/
I'll trigger another job from `anomaly_classification` branch

**Review by samet-akcay — APPROVED** [review]

**Review by djdameln — APPROVED** [review]

**Pull request merged**